addons/account/models/account_partial_reconcile.py

```diff
@@ -135,9 +136,21 @@ def unlink(self):
     @api.model_create_multi
     def create(self, vals_list):
         partials = super().create(vals_list)
+        self._pay_matched_payment(partials)
         self._update_matching_number(partials.debit_move_id + partials.credit_move_id)
         return partials
 
+    @api.model
+    def _pay_matched_payment(self, partials):
+        for partial in partials:
+            amount = partial.credit_amount_currency
+            matched_payments = partial.credit_move_id.move_id.matched_payment_ids | partial.debit_move_id.move_id.matched_payment_ids
```

Comment:
```suggestion
            matched_payments = (partial.credit_move_id | partial.debit_move_id).move_id.matched_payment_ids
```

Shorter to write, but also a few less SQL queries :wink: